butte-build/src/codegen.rs

```diff
@@ -101,6 +101,7 @@ fn to_type_token(
     ty: &ir::Type,
     lifetime: &impl ToTokens,
     wrap_refs_types: &impl ToTokens,
+    wrap_outer: bool,
```

Comment:
Just out of curiosity could we avoid adding this flag if we made `wrap_refs_types` into an `Option<&impl ToTokens>`?

Reply:
I'd be happy to do this in a follow up in the interest of merging ASAP.

Reply:
> Just out of curiosity could we avoid adding this flag if we made wrap_refs_types into an Option<&impl ToTokens>?

We couldn't, this helps with complex types like `Vector` or `WIPOffset` that are not wrapped the same as the "internal" type.  I'm not really happy with this logic in general, but first step is to make it work, then make it nice :)

Reply:
Absoultely, carry on.